packages/webapp/src/containers/Map/saga.js

```diff
@@ -162,10 +163,25 @@ export function* bulkUploadSensorsInfoFileSaga({ payload: { file } }) {
   }
 }
 
+export const getSensorReadings = createAction('getSensorReadingsSaga');
+
+export function* getSensorReadingsSaga() {
+  const { user_id, farm_id } = yield select(userFarmSelector);
+  const header = getHeader(user_id, farm_id);
+  try {
+    yield put(onLoadingSensorReadingStart(user_id, farm_id));
+    const result = yield call(axios.get, `${sensorUrl}/sensor_readings/${farm_id}/7`, header);
+    yield put(getSensorReadingSuccess(result.data));
+  } catch (e) {
+    console.error(e);
```

Comment:
can clear the state in case the API fails. It's better to clear the data so that If someone is using the same data in another component will not be showing wrong information using the selector function.

Reply:
Good idea! I only put the data in the store when the api returned correct data